### ISSUE TITLE
Gateway Functionality in v 0.3.2a hybrid

### ISSUE THREAD

**Issue by wolfravenous** [Byzantium/Byzantium#210]
I am booting an Asus EEE pc from a USB stick which I setup on a different computer running the system live from a cd.
Bootup seemed to go fine.
Graphics were recognized welll. 
Wlan was configured and is broadcasting.
I have been able to connect to the mesh from two different power books and use the webirc client to chat with my wife across the room, heehe.  

The one issue I seem to be having is on the host node, the Asus EEE machine, when it boots up the firefox screen that launches and presents home/guest/.passfail/success.html indicates "and if you plug a network cable into the appropriate jack this node will turn itself into a gateway for the rest of the mesh".  However, it doesn't seem to be establishing a gateway automagically.

I was attempting to do some troubleshooting on my own, and ipconfig on the node reveals that the NIC was recognized/configured by porteus. I can connect to the outside internet from the node, but not from machines that I connect to the node.

I was looking for the option to configure it in the control panel but wasn't getting the control panel on port 8080, then I read in a previous "issue" post that the control panel was removed as of version 0.3.  So I ran a netsat -l to see what other ports were listening.

I noticed 2006 and 9090 which I recognized from the boot up screen, when visiting 9090 it provided me with what appears to be configuration settings for  Byzantium.  Am not sure how to manual configure those settings to try to establish a gateway, for example, the settings 

"gateways": [
]

appears to be empty and also the hna setting which I am not sure if it is relevant at all??? has a setting

"hna": [
{
     "destination":  "0.0.0.0",
     "genmask" 0,
     "gateway": "192.168.128.206"}]

I did not see any mention of Eth0 which is the nic card that provides an internet connection on the node in the displayed file.

Is it possible to manually configure the gateway by editing this configuration file that is being served up on port 9090? If so can someone please tell me where the file is locate and what I should put into it to configure the gateway, then do I need to restart network services or something like that and what command would do that?  

Thanks in advance for assistance.

**Comment by wolfravenous** [comment]
Going to fall back down to version 0.2a to try and determine if I can set up the gateway using the manual control panel instead of auto configuration of the gateway that is currently failing for me in version 0.3.2a hybrid.  This may provide more feedback as to why the gateway isn't autoconfiguring.

**Comment by wolfravenous** [comment]
Same problem with version 0.2a  so I attempted to switch computers, use the mac laptop as node and asus eee as client to test gateway funtionality, but it still didn't work so it wasn't an issue unique to the asus eee

**Comment by wolfravenous** [comment]
Then after a discussion with haxwithaxe on irc we pondered that it could be because the ethernet cable hooked to the node was from a router using 10/8 parameters, and the mesh is also a 10/8 network and that the node machine possilbly couldn't tell how to route the requests.  This was correct, when I took the router out of the equation and hooked the mac laptop directly to the dsl modem it pulled an ip from the dsl modem of 192.168.x.x and the gateway functionality did work.  I am attempting further testing.  I want to put my router back in the equation and test using a 172.17.x.x network parameters and 192.168.x.x network parameters.

**Comment by wolfravenous** [comment]
Ok tested with the router back in the system between the dsl modem and the node.  The router was pulling the 192.168.x.x address from the dsl modem and it was handing out 172.17.x.x ip adresses to the node machine (mac laptop).  When I tried to connect from another laptop to the Byzantium mesh network a 10/8 I could connect but I didn't get any gateway functionality.  So I am going to do a final test with the router in the system  between the dsl modem and the node with it using 192.168 params and see what happens.

**Comment by wolfravenous** [comment]
Ok changed the router to 192.168 on the inside, then booted up the mac node and it pulled a 192.168 from the router, and was handing out 10.155 addresses on the mesh, and I still didn't have gateway functionality, so the issue seems to be entirely within the configurations of my router, maybe some dns settings in the router, not sure, but the node works as a gateway when connected directly to the dsl modem... Will have to play around with my router sometime when I get a chance and try to determine why it wasn't resolving the requests coming from the byz node, at this point I don't think it has anything to do with the byz gateway configs or anything that could be solved by altering them.  Thanks.

**Comment by wolfravenous** [comment]
The problem was that I had not entered the gateway and DNS setting under the DHCP section in my router.  These were listed as optional.  Under gateway they had:
"(Optional.) Suggest to input the IP Address of the LAN port of the Router"
So, I entered a gateway value of 172.17.1.1 which is my routers inside LAN address.
Under DNS Where the router description was:
Primary DNS - (Optional.) Input the DNS IP address provided by your ISP. Or consult your ISP.
Secondary DNS - (Optional.) Input the IP Address of another DNS server if your ISP provides two DNS servers.
I ignored my ISP and went with the OpenDNS settings of 208.67.222.222 and 208.67.220.220.  Now my Byzantium gateway functions with the home router in the equation :dancer:
